fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -1020,11 +1020,14 @@ public MetaDataEvolutionValidator getEvolutionValidator() {
      * in the meta-data proto.
      *
      * @param evolutionValidator the validator used to check the local file descriptor against the one in the meta-data proto
+     * @return this builder
      * @see #setLocalFileDescriptor(Descriptors.FileDescriptor)
      * @see MetaDataEvolutionValidator
      */
-    public void setEvolutionValidator(@Nonnull MetaDataEvolutionValidator evolutionValidator) {
+    @Nonnull
+    public RecordMetaDataBuilder setEvolutionValidator(@Nonnull MetaDataEvolutionValidator evolutionValidator) {
```

Comment:
> You definitely want the `evolutionValidator` set before you set `records`

Then how about you explicitly disallow it here?

Reply:
Good idea. Done.